core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -168,6 +168,10 @@ private static String implFromLocation(String location) {
     return SCHEME_TO_FILE_IO.getOrDefault(scheme(location), FALLBACK_IMPL);
   }
 
+  public Class<? extends FileIO> ioClass(String location) {
+    return io(location).getClass();
```

Comment:
can we get the class from `implFromLocation` through `Class.forName(className)`? `io()` will do the actual loading which is more expensive

Reply:
we also wanted to check if any fileIO extending HadoopFileIO should also be covered hence didn't use the implFromFileLocation, but definitely loading S3FileIO etc is going to be expensive, was thinking we would eventually need the fileIO obj anyway since we check the table location at the moment (unless the location of metadata and data path are overidden by table prop), Should we restrict the check to only HadoopFileIO, Your thoughs ?

Reply:
Sorry did not fully get what you mean here, if a FileIO extends HadoopFileIO, we can still get its class in this way and then compare it later to see if it is subclass right?

Reply:
Apologies, i see now what you meant, let me make the changes.